lib/tapioca/runtime/trackers/mixin.rb

```diff
@@ -20,26 +21,35 @@ class Type < T::Enum
         sig do
           params(
             constant: Module,
-            mod: Module,
+            mixin: Module,
             mixin_type: Type,
             locations: T.nilable(T::Array[Thread::Backtrace::Location])
           ).void
         end
-        def self.register(constant, mod, mixin_type, locations)
+        def self.register(constant, mixin, mixin_type, locations)
           locations ||= []
           locations.map!(&:absolute_path).uniq!
+
           locs = mixin_locations_for(constant)
-          locs.fetch(mixin_type).store(mod, T.cast(locations, T::Array[String]))
+          locs.fetch(mixin_type).store(mixin, T.cast(locations, T::Array[String]))
+
+          constants = constants_with_mixin(mixin)
+          constants << (constant)
         end
 
         sig { params(constant: Module).returns(T::Hash[Type, T::Hash[Module, T::Array[String]]]) }
         def self.mixin_locations_for(constant)
-          @mixin_map[constant] ||= {
+          @constants_to_mixin_locations[constant] ||= {
             Type::Prepend => {}.compare_by_identity,
             Type::Include => {}.compare_by_identity,
             Type::Extend => {}.compare_by_identity,
           }
         end
+
+        sig { params(mixin: Module).returns(T::Array[Module]) }
```

Comment:
Keeping this hash super simple for right now, but I think we could add the actual locations of the mixins in a follow-up PR.